src/Cli/dotnet/commands/dotnet-workload/repair/WorkloadRepairCommand.cs

```diff
@@ -0,0 +1,118 @@
+            var sdkFeatureBand = new SdkFeatureBand(_sdkVersion);
+            tempDirPath = tempDirPath ?? (string.IsNullOrWhiteSpace(parseResult.ValueForOption<string>(WorkloadInstallCommandParser.TempDirOption)) ?
+                Path.GetTempPath() :
+                parseResult.ValueForOption<string>(WorkloadInstallCommandParser.TempDirOption));
+            var tempPackagesDir = new DirectoryPath(Path.Combine(tempDirPath, "dotnet-sdk-advertising-temp"));
+            nugetPackageDownloader = nugetPackageDownloader ?? new NuGetPackageDownloader(tempPackagesDir, filePermissionSetter: null, new NullLogger());
+            _workloadInstaller = workloadInstaller ?? 
+                WorkloadInstallerFactory.GetWorkloadInstaller(_reporter, sdkFeatureBand, _workloadResolver, _verbosity, nugetPackageDownloader, dotnetDir, tempDirPath, _packageSourceLocation);
+        }
+
+        public override int Execute()
+        {
+            try
+            {
+                _reporter.WriteLine();
+
+                var workloadIds = _workloadInstaller.GetWorkloadInstallationRecordRepository().GetInstalledWorkloads(new SdkFeatureBand(_sdkVersion));
+
+                _reporter.WriteLine(string.Format(LocalizableStrings.RepairingWorkloads, string.Join(" ", workloadIds)));
+
+                ReinstallWorkloadsBasedOnCurrentManifests(workloadIds, new SdkFeatureBand(_sdkVersion));
+
+                if (_workloadInstaller.GetInstallationUnit().Equals(InstallationUnit.Packs))
+                {
+                    _workloadInstaller.GetPackInstaller().GarbageCollectInstalledWorkloadPacks();
+                }
+
+                _reporter.WriteLine();
+                _reporter.WriteLine(string.Format(LocalizableStrings.RepairSucceeded, string.Join(" ", workloadIds)));
+                _reporter.WriteLine();
+            }
+            catch (Exception e)
+            {
+                // Don't show entire stack trace
+                throw new GracefulException(string.Format(LocalizableStrings.WorkloadRepairFailed, e.Message), e);
+            }
+
+            return 0;
+        }
+
+        private void ReinstallWorkloadsBasedOnCurrentManifests(IEnumerable<WorkloadId> workloadIds, SdkFeatureBand sdkFeatureBand)
+        {
+            if (_workloadInstaller.GetInstallationUnit().Equals(InstallationUnit.Packs))
+            {
+                var installer = _workloadInstaller.GetPackInstaller();
+
+                var packsToInstall = workloadIds
+                    .SelectMany(workloadId => _workloadResolver.GetPacksInWorkload(workloadId.ToString()))
+                    .Distinct()
+                    .Select(packId => _workloadResolver.TryGetPackInfo(packId))
+                    .Where(pack => pack != null);
+
+                foreach (var packId in packsToInstall)
+                {
+                    installer.InstallWorkloadPack(packId, sdkFeatureBand);
```

Comment:
This will run the following underneath. Will the following works when there is existing files? File.Copy would throw when there are existing files at least. I think existing tests cannot validate that since they are mocks
```c#

if (IsSingleFilePack(packInfo))
                            {
                                File.Copy(packagePath, packInfo.Path);
                            }
                            else
                            {
                                var tempExtractionDir = Path.Combine(_tempPackagesDir.Value, $"{packInfo.Id}-{packInfo.Version}-extracted");
                                tempDirsToDelete.Add(tempExtractionDir);
                                Directory.CreateDirectory(tempExtractionDir);
                                var packFiles = _nugetPackageDownloader.ExtractPackageAsync(packagePath, new DirectoryPath(tempExtractionDir)).Result;

                                FileAccessRetrier.RetryOnMoveAccessFailure(() => Directory.Move(tempExtractionDir, packInfo.Path));
                            }
```

Reply:
It checks `if (!PackIsInstalled(packInfo))` on line 87, so it won't get to this logic if there are existing files.